File: 2024/day9/python/test_problem.py
import unittest
from problem import problem_part1, problem_part2

class TestProblem(unittest.TestCase):
    def test_problem_part1_test1(self):
        input_list = [
            "12345"
        ]
        result = problem_part1(input_list)
        self.assertEqual(result, 60)


    def test_problem_part1_test2(self):
        input_list = [
            "2333133121414131402"
        ]
        result = problem_part1(input_list)
        self.assertEqual(result, 1928)


    def test_problem_part2_test1(self):
        input_list = [
            "2333133121414131402"
        ]
        result = problem_part2(input_list)
        self.assertEqual(result, 2858)

File: 2025/day6/julia/TestProblem.jl
using Test

include("Problem.jl")

@testset "Problem" begin
    input_vector = [
        "123 328  51 64 ",
        " 45 64  387 23 ",
        "  6 98  215 314",
        "*   +   *   +  "
    ]
    @test problem_part1(input_vector) == 4277556

    @test problem_part2(input_vector) == 3263827
end